src/ray/common/task/task_spec.cc

```diff
@@ -53,12 +53,23 @@ void TaskSpecification::ComputeResources() {
 
 // Task specification getter methods.
 TaskID TaskSpecification::TaskId() const {
+  if (message_->task_id().empty()) {
```

Comment:
When are these used? Is it just an optimization? Might be good to leave a comment

Reply:
This otherwise can cause crashes when deserializing from empty protos in tests. Not strictly required, but is a slightly nicer behaviour.